app/businesses/uploads.py

```diff
@@ -0,0 +1,158 @@
+import csv
+from datetime import time
+
+from flask import current_app
+from sqlalchemy.exc import SQLAlchemyError
+
+from app.businesses.models import Business, Phone, BusinessHour, Address, SocialLink, Tag, BusinessUpload
+from app.businesses.repositories import BusinessUploadRepository, TagRepository
+
+upload_repository = BusinessUploadRepository()
+tag_repository = TagRepository()
+
+
+def process_file(filename):
+    upload = BusinessUpload()
+    try:
+        businesses = extract_business_from_csv(filename)
+        upload.addBusinesses(businesses)
+        upload.success = True
+        upload.filename = filename
+        upload_repository.save(upload)
+    except (Exception,SQLAlchemyError) as e:
+        upload.businesses = []
+        upload.success = False
+        upload_repository.save(upload)
+        upload.error_message = str(e.args[0])
+        current_app.logger.error(str(e.args[0]))
```

Comment:
si cette parti contient des exception, elle seront catch dans le function qui l'appel dans un try and except